src/app/app.component.ts

```diff
@@ -1,20 +1,62 @@
-import {Component, OnInit} from '@angular/core';
-import {Router} from '@angular/router';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { Link } from './app.api';
+import { NavigationService } from './navigation.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.css']
+  styleUrls: ['./app.component.css'],
 })
-export class AppComponent implements OnInit {
-  constructor(private readonly router: Router) {
+export class AppComponent implements OnInit, OnDestroy {
+  /**
+   * List of all valid links
+   */
+  links: Link[];
+
+  /**
+   * For easier handling of the navigation via the button the index of
+   * the currently selected index is used
+   */
+  selectedLinkIndex: number;
+
+  private subscription: Subscription;
+
+  constructor(private navigationService: NavigationService, private router: Router) {
+    this.links = [];
   }
 
-  public ngOnInit(): void {
+  ngOnInit(): void {
+    this.subscription = this.navigationService.getNavigationItems().subscribe((links) => {
+      if (links.length && this.selectedLinkIndex === undefined) {
+        this.selectedLinkIndex = 0;
+      }
+      this.links = links;
+    });
```

Comment:
Da wir die Daten im Template verwenden, können wir auch direkt das Observable an das Template geben. Änderungen im Template merke ich auch an.
```suggestion
    this.links$ = this.navigationService.getNavigationItems();
```